server/game/core/ability/abilityTargets/CardTargetResolver.ts

```diff
@@ -85,13 +85,16 @@ export class CardTargetResolver extends TargetResolver<ICardTargetResolver<Abili
 
         // if there are legal targets but this wouldn't have a gamestate-changing effect on any of them, we can just shortcut and skip selection
         // (unless there are dependent targets that might care about the targeting result)
+        // also, some complex implementations(e.g. Don't Get Cocky) use a targetResolver with no immediateEffect to have the player choose a target for referencing later
+        // these will appear to have no effect on any target, but should not be skipped
         if (
             !this.dependentTarget &&
+            this.immediateEffect &&
             !legalTargets.some((target) => this.immediateEffect.canAffect(
                 target,
                 this.getContextCopy(target, context),
                 {},
-                GameStateChangeRequired.MustFullyOrPartiallyResolve
+                this.properties.mustChangeGameState
```

Comment:
@AMMayberry1 Should I change this line back to how it was and delete the default setting in the `TargetResolver` constructor or is it better this way?

Reply:
I think it should go back to the way it was for now, unless we find a case where it really needs to be configurable